packages/celotool/src/lib/geth.ts

```diff
@@ -614,3 +639,501 @@ export const transferERC20Token = async (
+  const admin = new Admin(enodeRpcUrl)
+
+  let nodeInfo: any = {
+    enode: null,
+  }
+
+  try {
+    nodeInfo = await admin.getNodeInfo()
+  } catch {
+    console.error(`Unable to get node info from ${enodeRpcUrl}`)
+  }
+
+  return nodeInfo.enode
+}
+
+export async function addStaticPeers(datadir: string, peers: string[], verbose: boolean) {
+  const staticPeersPath = `${datadir}/static-nodes.json`
+  if (verbose) {
+    console.log(`Writing static peers to ${staticPeersPath}`)
+  }
+
+  const enodes = await Promise.all(peers.map((peer) => getEnode(peer)))
+  const enodesString = JSON.stringify(enodes, null, 2)
+
+  if (verbose) {
+    console.log('eNodes', enodesString)
+  }
+
+  fs.writeFileSync(staticPeersPath, enodesString)
+}
+
+export async function addProxyPeer(
+  runPath: string,
+  gethBinaryPath: string,
+  instance: GethInstanceConfig
+) {
+  if (instance.proxies) {
+    await spawnCmdWithExitOnFailure(gethBinaryPath, [
+      '--datadir',
+      getDatadir(runPath, instance),
+      'attach',
+      '--exec',
+      `istanbul.addProxy('${instance.proxies[0]!}', '${instance.proxies[1]!}')`,
+    ])
+  }
+}
+
+export async function startGeth(
+  gethConfig: GethRunConfig,
+  gethBinaryPath: string,
+  instance: GethInstanceConfig,
+  verbose: boolean
+) {
+  if (verbose) {
+    console.log('starting geth with config', JSON.stringify(instance, null, 2))
+  } else {
+    console.log(`${instance.name}: starting.`)
+  }
+
+  const datadir = getDatadir(gethConfig.runPath, instance)
+
+  const {
+    syncmode,
+    port,
+    rpcport,
+    wsport,
+    validating,
+    validatingGasPrice,
+    bootnodeEnode,
+    isProxy,
+    proxyAllowPrivateIp,
+    isProxied,
+    proxyport,
+    ethstats,
+  } = instance
+
+  const privateKey = instance.privateKey || ''
+  const lightserv = instance.lightserv || false
+  const etherbase = instance.etherbase || ''
+  const verbosity = gethConfig.verbosity ? gethConfig.verbosity : '3'
+  let blocktime: number = 1
+
+  if (
+    gethConfig.genesisConfig &&
+    gethConfig.genesisConfig.blockTime !== undefined &&
+    gethConfig.genesisConfig.blockTime >= 0
+  ) {
+    blocktime = gethConfig.genesisConfig.blockTime
+  }
+
+  const gethArgs = [
+    '--datadir',
+    datadir,
+    '--syncmode',
+    syncmode,
+    '--debug',
+    '--port',
+    port.toString(),
+    '--rpcvhosts=*',
+    '--networkid',
+    gethConfig.networkId.toString(),
+    `--verbosity=${verbosity}`,
+    '--consoleoutput=stdout', // Send all logs to stdout
+    '--consoleformat=term',
+    '--nat',
+    'extip:127.0.0.1',
+  ]
+
+  if (rpcport) {
+    gethArgs.push(
+      '--rpc',
+      '--rpcport',
+      rpcport.toString(),
+      '--rpccorsdomain=*',
+      '--rpcapi=eth,net,web3,debug,admin,personal,txpool,istanbul'
+    )
+  }
+
+  if (wsport) {
+    gethArgs.push(
+      '--wsorigins=*',
+      '--ws',
+      '--wsport',
+      wsport.toString(),
+      '--wsapi=eth,net,web3,debug,admin,personal'
```

Comment:
Might as well expose the same APIs over WS
```suggestion
      '--wsapi=eth,net,web3,debug,admin,personal,txpool,istanbul'
```